src/main/java/org/commcare/suite/model/Entry.java

```diff
@@ -140,11 +143,12 @@ public String getDisplayText() {
     }
 
     @Override
-    public String getTextForBadge(EvaluationContext ec) {
+    public Observable<String> getTextForBadge(EvaluationContext ec) {
         if (display.getBadgeText() == null) {
-            return null;
+            return Observable.just("");
         }
-        return display.getBadgeText().evaluate(ec);
+        return Observable.just(display.getBadgeText().evaluate(ec))
+                .subscribeOn(Schedulers.computation());
```

Comment:
Great catch. It shouldn't be here as well as I am letting the caller decide(Android code) which Scheduler to use. Will change it.